Support Test Resources from @TestProfile in native mode

Sgitario opened the pull request:
## Description
In https://github.com/quarkusio/quarkus/pull/13154, the annotation `@TestProfile` was supported also in Native tests. However, the native extension was not processing the test resources in the test profile like:

```
public class ConfluentTestProfile implements QuarkusTestProfile {

    @Override
    public String getConfigProfile() {
        return "confluent";
    }

    @Override
    public List<TestResourceEntry> testResources() {
        return Collections.singletonList(new TestResourceEntry(ConfluentKafkaResource.class));
    }
}
```

This PR makes the above to be supported.
The solution behaves the same as done in the Quarkus Test extension.

Fix https://github.com/quarkusio/quarkus/issues/18072

Sgitario commented:
@geoand I provided a possible fix for this issue: https://github.com/quarkusio/quarkus/issues/18072
Feel free to scrach the solution in the attached PR if you prefer another approach or see any drawbacks.
I tried these changes with using testcontainers as well and it works fine now.

Sgitario commented on test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeTestExtension.java:
```diff
@@ -175,6 +179,39 @@ public void postProcessTestInstance(Object testInstance, ExtensionContext contex
+     * {@link TestResourceManager.TestResourceClassEntry}
+     * that are loaded from that ClassLoader
+     */
+    private List<TestResourceManager.TestResourceClassEntry> getAdditionalTestResources(
```
This code is the same as in QuarkusTestExtension: https://github.com/quarkusio/quarkus/blob/a5c0d7710d677462a92855ba5948c2876f18260e/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java#L435

As the logic inside is not very complicated, I decided to not refactor both extensions and reuse the same method, but let me know if you prefer me doing so.

geoand commented:
I'll check it out tomorrow, thanks.

In the meantime, there are failing JVM tests

Sgitario committed:
Support Test Resources from @TestProfile in native mode

## Description
In https://github.com/quarkusio/quarkus/pull/13154, the annotation `@TestProfile` was supported also in Native tests. However, the native extension was not processing the test resources in the test profile like:

```
public class ConfluentTestProfile implements QuarkusTestProfile {

    @Override
    public String getConfigProfile() {
        return "confluent";
    }

    @Override
    public List<TestResourceEntry> testResources() {
        return Collections.singletonList(new TestResourceEntry(ConfluentKafkaResource.class));
    }
}
```

This PR makes the above to be supported.
The solution behaves the same as done in the Quarkus Test extension.

Sgitario commented:
> I'll check it out tomorrow, thanks.
> 
> In the meantime, there are failing JVM tests

I've just fixed the failing tests.

geoand reviewed (approved):
LGTM, thanks!
